app/models/edition/timestamps.rb

```diff
@@ -0,0 +1,58 @@
+class Edition::Timestamps
+  def self.edited(
+    edition,
+    payload,
+    previous_live_version = nil,
+    now = Time.zone.now
+  )
+    edition.temporary_first_published_at = previous_live_version&.first_published_at
+    edition.temporary_last_edited_at = now
+    edition.publisher_first_published_at = payload[:first_published_at]
+
+    unless edition.update_type == "major"
+      edition.major_published_at = previous_live_version&.major_published_at
+    end
+
+    edition.first_published_at = payload[:first_published_at] ||
+      previous_live_version&.first_published_at
+
+    edition.last_edited_at = payload[:last_edited_at] || now
+
+    edition.public_updated_at = payload[:public_updated_at] ||
+      previous_live_version&.public_updated_at
+
+    edition.publisher_major_published_at = payload[:public_updated_at]
+    edition.publisher_published_at = nil
```

Comment:
I'm guessing we might not need this set to nil as it will default to that?

Reply:
I know it was probably me who did that in first place too heh

Reply:
I did wonder, and thought maybe it's worth making it explicit in the code that we do want it to be nil.  Equally, I think the tests should make that clear, so... *shrugs*

Reply:
Yeah I think best to just remove it. Leaving it in only makes you wonder what it might have been.